internal/rbd/snapshot.go

```diff
@@ -145,15 +145,29 @@ func (rbdSnap *rbdSnapshot) toVolume() *rbdVolume {
 }
 
 func (rbdSnap *rbdSnapshot) ToCSI(ctx context.Context) (*csi.Snapshot, error) {
+	created, err := rbdSnap.GetCreationTime(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	ready, err := rbdSnap.GetReadyToUse(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	return &csi.Snapshot{
 		SizeBytes:      rbdSnap.VolSize,
 		SnapshotId:     rbdSnap.VolID,
 		SourceVolumeId: rbdSnap.SourceVolumeID,
-		CreationTime:   timestamppb.New(*rbdSnap.CreatedAt),
-		ReadyToUse:     true,
+		CreationTime:   timestamppb.New(*created),
+		ReadyToUse:     ready,
 	}, nil
 }
 
+func (rbdSnap *rbdSnapshot) GetReadyToUse(ctx context.Context) (bool, error) {
```

Comment:
Seems like this function always return true??

Reply:
+1
Will there be a newer definition of ReadyToUse with groupSnapshot ?

Reply:
Yes, currently it always returns `true`.

It is important for the VolumeGroupSnapshot to be able to see if the Snapshot is ready. Only when all snapshots are ready, the VolumeGroupSnapshot can be ready too. I did use that check in the implementation earlier,  but have removed it again as I'm now checking on other things too.

If you prefer, I'll drop this function from the interface too (and from the VolumeGroupSnapshot interface).

Reply:
> Yes, currently it always returns `true`.
> 
> It is important for the VolumeGroupSnapshot to be able to see if the Snapshot is ready. Only when all snapshots are ready, the VolumeGroupSnapshot can be ready too. I did use that check in the implementation earlier, but have removed it again as I'm now checking on other things too.
> 
> If you prefer, I'll drop this function from the interface too (and from the VolumeGroupSnapshot interface).

Yes, Let's drop it until we have a use case for the function and a safe way to use `ReadyToUse`.
I remember cephcsi stopped using `ReadyToUse` since we had stale resource issue earlier.

refer https://github.com/ceph/ceph-csi/pull/2260/commits/0ddd2d459f4e1684c93604bcc54a49c0791aa5aa